### DISCUSSION TITLE
run fastapi and access it via http ?

### DISCUSSION THREAD

**Discussion by spxneo** [pyodide/pyodide#4618] [General]
I'm curious to know what packages are usable. For instance, is it possible to run fastapi and access it locally? what about exposing that to a trusted remote server for bidrectional communication?

**Comment by hoodmane** [comment]
It's possible to run fastapi but you need a custom asgi server shim, for instance here:
https://github.com/cloudflare/workerd/blob/main/src/pyodide/internal/asgi.py
If you use a JavaScript server such that handlers receive a `Request` and return a `Response` then that can adapt so that you can respond to the requests via fastapi.

**Comment by spxneo** [comment]
all I need is a http server, is there something that is more user friendly or does it still require custom asgi shim
